### PR TITLE
[#27] Delete device in frontend

### LINKED ISSUE #27: Frontend: As a user, I want to be able to delete a device
**Acceptance Criteria**

When viewing detailed information of a device
- A delete button, like a trash can, should be shown somewhere on the page
- When clicking that button, a popup appears to confirm with the user that they really want to delete the device
- If they click cancel, the popup closes and nothing happens
- If they click OK, a DELETE request is sent to the server to delete the device, and the user is returned to the device list page

Acceptance test is #227

### PR DESCRIPTION
**Pull request by gdzooks**
# General info
**Issue number**: #27 

**Task description**: 
- A delete button, like a trash can, should be shown somewhere on the page
- When clicking that button, a popup appears to confirm with the user that they really want to delete the device
- If they click cancel, the popup closes and nothing happens
- If they click OK, a DELETE request is sent to the server to delete the device, and the user is returned to the device list page
 
 
| Original   | Updated|
|:---------------:|:---------------:|
| image | image |

# Testing
**Test coverage**:
- 87%

# Additional notes

**Note to reviewers**:
Once again counting on you guys to test that the `axios.delete` works as it should in the backend xD

**To do before merging**:
- [x] Fix Tests

### CODE REVIEW

**Review by felixlapierre — COMMENTED** [review]
Some feedback on the functionality:

1. The delete button in the device list page has no functionality (can be added in a later pr)
2. After confirming the delete, the user is not redirected to the device list page as specified in the requirements. This can be filed as a bug if we want and done in another PR, but i'd like to have it.
3. You're calling the delete encoder/decoder endpoint, which does delete the enc/dec, but the "device" entity in the database remains, meaning i can't re-register the device without manually deleting in postman. Instead, call the delete device endpoint (this might be a problem with my instructions i gave). That will cause problems when a device has encoder and decoder, since both will be deleted, but that's fine. We will have to address that problem in the backend.

TL;DR: Call the delete device endpoint instead. Fix 1 & 2 if you have time.

Lookin' at the code now.

**Comment by gdzooks** [comment]
> TL;DR: Call the delete device endpoint instead. Fix 1 & 2 if you have time.

changed endpoint call in 04e7f88!
lemme know if it works! i'll start on 1&2 in the meantime :,3

**edit:** finished 2 with thanks to @arejayelle ! 3f4660e :heart:
will leave the delete on the devicelist page for another PR

**Comment by felixlapierre** [comment]
Will merge this once the merge conflicts are fixed @gdzooks 🛩️ 😄

**Review by felixlapierre — DISMISSED** [review]

File: frontend/src/devicelist/ActionMenu.jsx [hunk 1]
```diff
@@ -26,6 +28,11 @@ export default class ActionMenu extends React.Component {
     this.setAnchorElement(null);
   }
 
+  handleDelete(deleteId) {
+    DeviceApi.deleteDevice(deleteId);
+    this.handleClose();
+  }
+
```
Comment (sharon-ho) [inline frontend/src/devicelist/ActionMenu.jsx]: Is this setup for Delete Device via the Device List page that's coming in a future PR?

**Review by sharon-ho — DISMISSED** [review]
Everything looks good! Just a small comment about some changed code in ActionMenu.

File: frontend/src/devicelist/ActionMenu.jsx (hunk 1 above)
Comment (gdzooks) [reply frontend/src/devicelist/ActionMenu.jsx]: ou yeah it is! i was going to try to do the delete button in this PR but then realized getting the confirmation dialog to work as well was gonna keep this issue open for too long

Comment (gdzooks) [reply frontend/src/devicelist/ActionMenu.jsx:35]: ou yeah it is!

Comment (sharon-ho) [reply frontend/src/devicelist/ActionMenu.jsx]: I see, so should this change be removed and added to that future PR you're mentioning?

Comment (gdzooks) [reply frontend/src/devicelist/ActionMenu.jsx]: i can remove it if you and @felixlapierre are willing to re-approve today :eyes:

Comment (sharon-ho) [reply frontend/src/devicelist/ActionMenu.jsx]: I can re-approve today for sure.

**Review by sharon-ho — APPROVED** [review]
LGTM! Looks like removing the unused code mentioned previously helped raise code coverage! 🎉

**Review by felixlapierre — APPROVED** [review]
Re-approve!
